### PR TITLE
Replace specs2 with hedgehog

### PR DESCRIPTION
**Pull request by Kevin-Lee**
## Replace specs2 with hedgehog
* Added `Gens` to replace `Arbitraries`
* Test code has been rewritten using Hedgehog
* Specs2 and ScalaCheck were removed from `libraryDependencies`

### CODE REVIEW

File: src/test/scala/pirate.spec/Laws.scala [hunk 1]
```diff
@@ -1,205 +1,462 @@
...
   }
 
   object plusEmpty {
-    def leftPlusIdentity[F[_], X](implicit f: PlusEmpty[F], afx: Arbitrary[F[X]], ef: Equal[F[X]]) =
-      forAll(f.plusEmptyLaw.leftPlusIdentity[X] _)
+    def leftPlusIdentity[F[_], A](genFA: Gen[F[A]])(implicit f: PlusEmpty[F], ef: Equal[F[A]]): Property =
+      for {
+        fa <- genFA.log("fa")
+      } yield {
+        f.plusEmptyLaw.leftPlusIdentity[A](fa) ==== true
```
Comment (charleso) [inline src/test/scala/pirate.spec/Laws.scala]: Super nit but prefer `Result.assert`

**Comment by charleso** [comment]
👍 

You mind rebasing?

File: src/test/scala/pirate.spec/Laws.scala (hunk 1 above)
Comment (Kevin-Lee) [reply src/test/scala/pirate.spec/Laws.scala]: OK, I will.

Comment (charleso) [reply src/test/scala/pirate.spec/Laws.scala]: Actually don't bother. I really don't care that much. :)

**Comment by charleso** [comment]
Let me know when you're ready to merge

**Comment by Kevin-Lee** [comment]
@charleso OK, thanks!

File: src/test/scala/pirate.spec/Laws.scala (hunk 1 above)
Comment (Kevin-Lee) [reply src/test/scala/pirate.spec/Laws.scala]: Already done. 🤣

**Comment by Kevin-Lee** [comment]
@charleso Ready to merge.

**Comment by charleso** [comment]
Nice one Kev!

**Pull request merged**